flink-kubernetes/src/main/java/org/apache/flink/kubernetes/utils/KubernetesUtils.java

```diff
@@ -0,0 +1,291 @@
+		File file = new File(filePath);
+		if (file.exists()) {
+			StringBuilder content = new StringBuilder();
+			String line;
+			try (BufferedReader reader = new BufferedReader(new InputStreamReader(new FileInputStream(file)))){
+				while ((line = reader.readLine()) != null) {
+					content.append(line).append(System.lineSeparator());
+				}
+			} catch (IOException e) {
+				throw new RuntimeException("Error read file content.", e);
+			}
+			return content.toString();
+		}
+		throw new FileNotFoundException("File " + filePath + " not exists.");
+	}
+
+	/**
+	 * Generates the shell command to start a job manager for kubernetes.
+	 * @param flinkConfig The Flink configuration.
+	 * @param jobManagerMemoryMb JobManager heap size.
+	 * @param configDirectory The configuration directory for the flink-conf.yaml
+	 * @param logDirectory The log directory.
+	 * @param hasLogback Uses logback?
+	 * @param hasLog4j Uses log4j?
+	 * @param mainClass The main class to start with.
+	 * @param mainArgs The args for main class.
+	 * @return A String containing the job manager startup command.
+	 */
+	public static String getJobManagerStartCommand(
+		Configuration flinkConfig,
+		int jobManagerMemoryMb,
+		String configDirectory,
+		String logDirectory,
+		boolean hasLogback,
+		boolean hasLog4j,
+		String mainClass,
+		@Nullable String mainArgs) {
+
+		final Map<String, String> startCommandValues = new HashMap<>();
+		startCommandValues.put("java", "$JAVA_HOME/bin/java");
+		startCommandValues.put("classpath", "-classpath " + "$" + Constants.ENV_FLINK_CLASSPATH);
+
+		startCommandValues.put("jvmmem", String.format("-Xms%dm -Xmx%dm", jobManagerMemoryMb, jobManagerMemoryMb));
+
+		startCommandValues.put("jvmopts", getJavaOpts(flinkConfig, CoreOptions.FLINK_JM_JVM_OPTIONS));
+
+		startCommandValues.put("logging",
+			getLogging(logDirectory + "/jobmanager.log", configDirectory, hasLogback, hasLog4j));
+
+		startCommandValues.put("class", mainClass);
+
+		startCommandValues.put("args", mainArgs != null ? mainArgs : "");
+
+		startCommandValues.put("redirects",
+			"1> " + logDirectory + "/jobmanager.out " +
+			"2> " + logDirectory + "/jobmanager.err");
+
+		final String commandTemplate = flinkConfig.getString(KubernetesConfigOptions.CONTAINER_START_COMMAND_TEMPLATE);
+		return BootstrapTools.getStartCommand(commandTemplate, startCommandValues);
+	}
+
+	/**
+	 * Generates the shell command to start a task manager for kubernetes.
+	 * @param flinkConfig The Flink configuration.
+	 * @param tmParams Parameters for the task manager.
+	 * @param configDirectory The configuration directory for the flink-conf.yaml
+	 * @param logDirectory The log directory.
+	 * @param hasLogback Uses logback?
+	 * @param hasLog4j Uses log4j?
+	 * @param mainClass The main class to start with.
+	 * @param mainArgs The args for main class.
+	 * @return A String containing the task manager startup command.
+	 */
+	public static String getTaskManagerShellCommand(
```

Comment:
Should it be `getTaskManagerStartCommand` to be aligned with `getJobManagerStartCommand `?

Reply:
Make sense.